lib/services/parser.go

```diff
@@ -192,7 +206,82 @@ func NewWhereParser(ctx RuleContext) (predicate.Parser, error) {
 		},
 		GetIdentifier: ctx.GetIdentifier,
 		GetProperty:   GetStringMapValue,
-	})
+	}
+	return def
+}
+
+// WhereParserOpt is a function that modifies the default
+// predicate.Def used to create a parser for the `where` section in access rules.
+type WhereParserOpt func(RuleContext, *predicate.Def)
+
+// WithCanViewFunction adds a can_view function to the parser definition.
+// This function will be used to check if the user has access to the resource
```

Comment:
I'm not sure I understand what `can_view` means. Is that separate from "can access" ??

Reply:
I initially named it as `can_access` but @Tener seemed confused by the terminology given that it doesn't really grant access to the resource - it only matched the resource labels. If a user doesn't have principals, he still doesn't have access to the resource so `can_access` seemed a bit incorrect.

IIRC, this is the terminology we use for label matching - can access the resource - but given that it caused some confusion, I renamed it to `can_view`. Properly implementing a `can_access` is quite difficult, specially for database access.

TL;DR `can_view` checks if the user role's match the server labels